syntax/types.rs

```diff
@@ -69,7 +69,9 @@ impl<'a> Types<'a> {
                 }
                 Api::CxxType(ety) => {
                     let ident = &ety.ident;
-                    if !type_names.insert(ident) {
+                    // We allow declaring the same type as a shared enum and as a Cxxtype, as this
+                    // means not to emit the C++ enum definition.
+                    if !type_names.insert(ident) && !enums.contains_key(ident) {
```

Comment:
I'd like for everything in Rust to stay order independent as much as possible, so please also deal with the Api::Enum being written after the Api::CxxType.

```rust
mod ffi {
    extern "C" {
        type E;
    }

    enum E {...}
}
```

It's fine to refactor how this duplication detection works if needed.

Reply:
Thanks for catching this!